src/use-swr-pages.tsx

```diff
@@ -205,7 +219,7 @@ export function useSWRPages<OffsetType = any, Data = any, Error = any>(
       p.push(pageCache[i].component)
     }
     return p
-  }, [_pageFn, pageCount, pageSWRs, pageOffsets, pageKey])
+  }, [_pageFn, pageCount, pageSWRs, pageOffsets, pageKey, pageOffsetKey])
```

Comment:
@innocentiv  `pageOffsetKey` will always change with `pageKey`... so it is not needed to put it here as a dependency, is it needed?

Reply:
It is not actually needed, but I think the recommended approach is to add all the values used inside the effect as a dependency to avoid stale closure pitfalls. There is none in this case, thanks to the link between `pageKey ` and `pageOffsetKey` , but I think it helps with code clarity and in future refactor. The link between these two values is elsewhere in the codebase, and the is no problem in adding a dependency. In general, there are lots of missing dependencies in swr. The issue was also discussed here: https://github.com/zeit/swr/issues/263 . We should try to decide the best way to manage dependencies in hooks.

More info here: https://github.com/facebook/react/issues/14920#issuecomment-471070149
dan abramov blog: https://overreacted.io/a-complete-guide-to-useeffect/#dont-lie-to-react-about-dependencies
react docs: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Reply:
I see... Thanks for clarifying it. Let's see if any of the reviewers decides to check the code soon. I've been waiting a lot this feature to be merged!